kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/CloudConnectionManager.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.cloudconnection;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloudconnection.publisher.CloudPublisher;
+import org.eclipse.kura.cloudconnection.subscriber.CloudSubscriber;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * The {@link CloudConnectionManager} provides an API layer to ease the communication with a remote server.
+ * {@link CloudConnectionManager} abstracts the developers from the complexity of the transport protocol and payload
+ * format used in the communication.
+ * It allows for a single connection to a remote server to be shared across more than one {@link CloudPublisher} and
+ * {@link CloudSubscriber} running in the gateway.
+ *
+ * Each cloud provider should provide its own {@link CloudConnectionManager} implementation in order to best fit the
+ * specific requirements.
+ *
+ * Applications should not use directly this API but, instead, use the {@link CloudPublisher} and the
+ * {@link CloudSubscriber} interfaces to give applications the capabilities to publish and receive messages.
+ *
+ * @noimplement This interface is not intended to be implemented by clients.
+ * @since 2.0
+ */
+@ProviderType
+public interface CloudConnectionManager extends CloudConnectionListenerRegistry {
+
+    /**
+     * Connects the framework to the remote cloud server.
+     *
+     * @throws KuraException
+     *             if the connection operation fails
+     */
+    public void connect() throws KuraException;
```

Comment:
Just to be concise, you could remove the access modifier from all the methods in this interface.